File: prepare_data.js
const fs = require('fs');

function updateChildren(node) {
    // Check if the node has children
    if (node.hasOwnProperty('children')) {
        node.children.forEach(child => {
            // Check if the child has a 'link' key
            if (child.hasOwnProperty('link')) {
                const linkValue = child.link;
                const oldChildren = [...child.children];
                // Update the children key as required
                child.children = [{
                    "name": linkValue,
                    "symbol": "none",
                    "collapsed": false,
                    "label": {
                        "color": "black",
                        "fontStyle": "bold"
                    },
                    "children": oldChildren
                }];
            }
            // Recursively apply to each child's children
            updateChildren(child);
        });
    }
}

fs.readFile('./data/fair-mapping.json', 'utf8', (err, data) => {
    if (err) throw err;
    let jsonData = JSON.parse(data);

    // Apply the function to the root node
    updateChildren(jsonData);

    // Optionally, you can save the updated JSON data back to the file
    fs.writeFile('./dist/fair-mapping.json', JSON.stringify(jsonData, null, 4), 'utf8', (err) => {
        if (err) throw err;
        console.log('The file has been saved!');
    });
});

File: prepare_data.py
import json

def update_children(node):
    # Check if the node has children
    if 'children' in node:
        for _, child in enumerate(node['children']):
            # Check if the child has a 'link' key
            if 'link' in child:
                link_value = child['link']
                old_children = child['children'].copy()
                # Update the children key as required
                child['children'] = [{
                    "name": link_value,
                    "symbol": "none",
                    "collapsed": False,
                    "label": {
                        "color": "black",
                        "fontStyle": "bold"
                    },
                    "children": old_children
                }]
            # Recursively apply to each child's children
            update_children(child)
if __name__=="__main__":
    with open("./data/fair-mapping.json", 'r') as _file:
        data = json.load(_file)
        # Apply the function to the root node
    update_children(data)

    # Optionally, you can save the updated JSON data back to the file
    with open('./dist/fair-mapping.json', 'w') as file:
        json.dump(data, file, indent=4)
